db/db_compaction_test.cc

```diff
@@ -1024,6 +1024,68 @@ TEST_F(DBCompactionTest, CompactionSstPartitioner) {
   ASSERT_EQ("B", Get("bbbb1"));
 }
 
+TEST_F(DBCompactionTest, CompactionSstPartitionWithManualCompaction) {
+  Options options = CurrentOptions();
+  options.compaction_style = kCompactionStyleLevel;
+  options.level0_file_num_compaction_trigger = 3;
+
+  DestroyAndReopen(options);
+
+  // create first file and flush to l0
+  ASSERT_OK(Put("000015", "A"));
+  ASSERT_OK(Put("000025", "B"));
+  ASSERT_OK(Flush());
+  ASSERT_OK(dbfull()->TEST_WaitForFlushMemTable());
+
+  ASSERT_OK(Put("000015", "A2"));
+  ASSERT_OK(Put("000025", "B2"));
+  ASSERT_OK(Flush());
+  ASSERT_OK(dbfull()->TEST_WaitForFlushMemTable());
+
+  // CONTROL 1: compact to Lmax without partitioner
```

Comment:
Does it matter that we need to compact to Lmax to prove the point? Also my debugger shows files[0]'s level = 1, not Lmax (assuming Lmax is the last level which is L6? in this case?).

If Lmax is needed to prove the point, can you assert the output file is in Lmax indeed?

Reply:
Lmax is irrelevant, so removing that part of the comment. 👍